ui/collapsible-panel.reel/collapsible-panel.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * @module ui/collapsible-panel.reel
+ * @requires montage/ui/component
+ */
+var Component = require("../component").Component,
+    PressComposer = require("../../composer/press-composer").PressComposer;
+
+/**
+ * @class CollapsiblePanel
+ * @extends Component
+ *
+ * A collapsible panel that can be expanded or collapsed with header and body
+ */
+exports.CollapsiblePanel = Component.specialize(/** @lends CollapsiblePanel# */ {
+    constructor: {
+        value: function CollapsiblePanel() {
+            this.super();
+            this._pressComposer = new PressComposer();
+        }
+    },
+
+    _pressComposer: {
+        value: null
+    },
+
+    /**
+     * Set this to true when clicking on header to collapse or expand a panel, only trgiiger animation at this stage.
+     * No animation when first time drawing
+     * @private
+     */
+    _isCollapsing: {
+        value: false
+    },
+
+    _isExpanded: {
+        value: null
+    },
+    /**
+     * Is the collapsible panel expanded or collapsed
+     * @type {boolean}
+     */
+    isExpanded: {
+        set: function (value) {
+            if (this._isExpanded == value) {
+                return;
+            }
```

Comment:
This would be more readable if instead of returning in the middle of the method (often a dangerous and harder to read thing) you instead reversed the test to only make the change if necessary. This could also be written even more simply as {this.needsDraw = (this._isExpanded != value); this._ixExpanded = value; }.